File: ApisREST/ApiStudent/app/Models/Course.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Course extends Model
{
    use HasFactory;
    protected $primaryKey = 'course_id';

    protected $fillable = ['title_course', 'description', 'sport_id', 'cdl_coach', 'quota', 'status'];

    public function sport()
    {
        return $this->belongsTo(Sport::class, 'sport_id');
    }

    public function coach()
    {
        return $this->belongsTo(User::class, 'cdl_coach');
    }

    public function students()
    {
        return $this->belongsToMany(User::class, 'course_student', 'course_id', 'cdl_student');
    }
}

File: ApisREST/ApiCourses/app/models/course.rb
class Course < ApplicationRecord
  self.primary_key = 'course_id'
  belongs_to :sport, foreign_key: 'sport_id', primary_key: 'sport_id'
  belongs_to :coach, class_name: 'User', foreign_key: 'cdl_coach', primary_key: 'cdl_user'

end
